### ISSUE TITLE
Combine softmax & triplet loss training

### ISSUE THREAD

**Issue by lyrgwlr** [KaiyangZhou/deep-person-reid#305]
I have trained the OSNet with your default config file on some datasets, and I found the result is better than the paper.The default loss is label-smooth softmax.  
Then I trained the same config except changing the loss to triplet. The result is worse. 
On market-1501:
** Results **
mAP: 44.6%
CMC curve
Rank-1  : 64.7%
Rank-5  : 83.3%
Rank-10 : 89.1%
Rank-20 : 93.1%
On duke:
** Results **
mAP: 37.2%
CMC curve
Rank-1  : 55.0%
Rank-5  : 73.4%
Rank-10 : 79.4%
Rank-20 : 83.6%
Why the triplet loss leeds to such bad results? And the "weight_t" and "weight_x" are provided in your code. So I wonder if the combination of softmax and triplet can develop the performance? Have you done some experiments on that?

**Comment by KaiyangZhou** [comment]
See table12 in https://arxiv.org/pdf/1905.00953.pdf

there are some discussions about triplet loss in the issue area, please do a search
